app/components/ApplicationPage/AuthPage/SSOCallbackPage/index.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { type Dispatch } from 'redux';
+import { flashErrorMessage } from 'redux-flash';
+import { translate, type TranslatorProps } from 'react-i18next';
+import { type ContextRouter as RouterProps } from 'react-router-dom';
+import { push } from 'connected-react-router';
+
+import { type Action } from 'types/action';
+import { InvalidArgumentError } from 'errors';
+import platform from 'modules/platform';
+import * as paths from 'config/routes';
+
+type DispatchProps = {|
+  signinSSO: (token: string, id: string) => void,
+  flashErrorAndRedirect: (error: string) => void,
+|};
+
+type Props = {| ...TranslatorProps, ...RouterProps, ...DispatchProps |};
+
+const mapDispatchToProps = (dispatch: Dispatch<Action>): DispatchProps => {
+  return {
+    signinSSO: (token: string, id: string): void => {
+      dispatch(platform.actions.signinSSO(token, id));
+    },
+    flashErrorAndRedirect: (error: string): void => {
+      dispatch(flashErrorMessage(error));
+      dispatch(push(paths.AUTH_SIGNIN_ROUTE));
```

Comment:
Side note: we really need a way to show error messages that aren't 'flashed', since it is possible for the user to never even see the message depending on the timing of these two actions. In the meantime, I think it would be better to just throw the error for now, so that at least the developer always gets feedback.

Reply:
Correct me if I'm wrong, but I thought that the flash messages worked with a timer, so what happens if the redirect is very slow?

Reply:
That doesn't mean it can't be slow; for example if the user is using a mobile device with low cpu power or is doing some kind of cpu intensive calculations in the background.

Reply:
Honestly, I guess the main issue is that I just don't like flash messages. I myself am too slow to read them half of the time, even if they're displayed on time (which I still don't think is guaranteed here). 

Typical scenario: [click button] -> [look down to sip tea] -> "Shit, something in red just appeared and disappeared on my screen and I have no idea what it said."

But that discussion doesn't really belong here, so just ignore this change request for now.

Reply:
If the user's device takes more than 5 seconds to render a new route within the bundle itself, it is nearly impossible for the user to have completed the OAuth step first, since it requires multiple redirects to an external provider and back, and the authorization itself. Not to mention that the rest of the platform will be unusable.

Show me a reproducible example for this edge case and I will look into changing it.

Reply:
Typically, flash messages are also persistent in the UI until the next redirect. However, the libraries I found that were suitable for our application implement flash messages using a timer.

Reply:
They're not though; that's why they're called "flash" messages :P Otherwise it'd just be a normal error message.

Reply:
I'm speaking mainly from my experience with full-stack solutions such as Ruby on Rails and derivative JS-based solutions. Flash messages are set and cleared using cookies, and an error box is rendered conditionally on that cookie. As you describe, the user could click a button, and go for a coffee, only to come back and see the flash message still on his/her screen.